eth/gasprice/feehistory.go

```diff
@@ -121,7 +121,19 @@ func (oracle *Oracle) processBlock(bf *blockFees, percentiles []float64) {
 
 	sorter := make(sortGasAndReward, len(bf.block.Transactions()))
 	for i, tx := range bf.block.Transactions() {
-		reward, _ := tx.EffectiveGasTip(bf.block.BaseFee())
+		var reward *big.Int
+		pendingTxCount, _ := oracle.backend.StatsWithMinBaseFee(headHeader.BaseFee)
+		if pendingTxCount < oracle.congestedThreshold {
+			// Before Curie (EIP-1559), we need to return the total suggested gas price. After Curie we return defaultGasTipCap wei as the tip cap,
+			// as the base fee is set separately or added manually for legacy transactions.
+			price := oracle.defaultGasTipCap
+			if !oracle.backend.ChainConfig().IsCurie(headHeader.Number) {
```

Comment:
this can be replaced by `bf.block.BaseFee()`, nil means Curie not enabled, non-nil means Curie enabled. then the added `headHeader` param can be removed, simplifying the logic.

Reply:
when `blockNumber` < pendingBlock.Number, then `fees.header, fees.err = oracle.backend.HeaderByNumber(ctx, rpc.BlockNumber(blockNumber))`, it isn't the current baseFee. so I don't use `bf.block.BaseFee()`

Reply:
https://github.com/scroll-tech/go-ethereum/blob/develop/eth/gasprice/feehistory.go#L273-L279

Reply:
make sense.